addon/ExperimentProvider.js

```diff
@@ -69,8 +110,15 @@ exports.ExperimentProvider = class ExperimentProvider {
   }
 
   destroy() {
-    simplePrefs.removeListener(OVERRIDE_PREF, this._onPrefChange);
-    this._data = {};
     this._experimentId = null;
+    Object.keys(this._experiments).forEach(experimentName => {
+      this._target.removeListener(experimentName, this._onPrefChange);
+    });
+  }
+
+  clearPrefs() {
```

Comment:
If I understand correctly, `destroy()` is called on any `unload` reason, 2 of which may be `disable` or `uninstall`. Perhaps another may be browser shut down? https://github.com/mozilla/activity-stream/blob/master/addon/ActivityStreams.js#L834

In any case. These prefs need to be persisted in all cases besides disable/uninstall which is why I made this a separate function that is only triggered in those cases.